code/__HELPERS/game.dm

```diff
@@ -214,17 +214,21 @@
 /proc/get_hearers_in_view(R, atom/source)
 	// Returns a list of hearers in view(R) from source (ignoring luminosity). Used in saycode.
 	var/turf/T = get_turf(source)
-	var/list/hear = list()
+	. = list()
 
 	if(!T)
-		return hear
-
-	var/list/range = get_hear(R, T)
-	for(var/atom/movable/A in range)
-		hear |= recursive_hear_check(A)
-
-	return hear
+		return
 
+	if (R == 0) // if the range is zero, we know exactly where to look for, we can skip view
+		. = recursive_hear_check(T)
+	else  // A variation of get_hear inlined here to take advantage of the compiler's fastpath for obj/mob in view
+		var/lum = T.luminosity
+		T.luminosity = 6

```

Comment:
Why 6?

Reply:
@MrStonedOne possibly knows, I have no idea. It's used in get_hear(), and presumably related to the slapfight in http://www.byond.com/forum/?post=1863682

Reply:
Because its the highest you can go to.
![](https://i.imgur.com/fUH1okI.png)

Reply:
Half the reason we do our own lighting shit is to bypass that bit of bullshit

Setting this above 6 used to make people grow dicks. (see #7510)